src/olympia/api/paginator.py

```diff
@@ -89,3 +89,23 @@ def get_paginated_response(self, data):
 class ESPageNumberPagination(CustomPageNumberPagination):
     """Custom pagination implementation to hook in our `ESPaginator`."""
     django_paginator_class = ESPaginator
+
+
+class OneOrZeroPageNumberPagination(CustomPageNumberPagination):
+    """Fake pagination that returns a result object like
+    CustomPageNumberPagination, but for special cases where we know we're never
+    going to have more than one object in the results anyway.
+    """
+    def paginate_queryset(self, queryset, request, view=None):
+        return list(queryset[:1])
```

Comment:
I don't know about a hard fail like an `assert`. I could add some logging maybe? Not sure it's worth it. I could envision this pagination class be used in a cases where you do have more than 1 object but you don't care.